app/controllers/scim_rails/scim_users_controller.rb

```diff
@@ -27,7 +27,13 @@ def index
     end
 
     def create
-      user = @company.public_send(ScimRails.config.scim_users_scope).create!(permitted_user_params)
+      username_key = ScimRails.config.queryable_user_attributes[:userName]
+      username_create_hash = Hash.new
```

Comment:
This update makes complete sense to me 👍 

My only suggestion would be to potentially name this something more appropriate - maybe `user_data`? I try to avoid using the data type in the name of a variable since it typically doesn't add a ton of clarity to the variable name.

I don't think this is something that's worth changing though since we need to get this out!

Reply:
I could see not using hash. This is not all the data though, it's only a key/value for finding or creating a user, the data is then applied with the permitted params in the next line. I could do something more descriptive like `find_by_username`. I need to make another small change so we can update this